### PR TITLE
Ballistics to Energy Part 2: Deltra and the Riot Blaster.

### PR DESCRIPTION
**Pull request by necromanceranne**
### Shotguns

- Replaces the majority of riot shotguns and the cargo purchased combat shotguns with the Riot Blaster Energy Shotgun. There is exactly one riot shotgun left in every maps armory. You can read about the Riot Blaster in the design document.

<details>
  <summary>...</summary>
  
I also removed DRAGnets from the armory and cargo.

It's okay though, I made their effects into energy shells. They're technically not gone, just the weapon itself.
  
</details>

- Most of the security shotgun shells in the lathe have been replaced with energy shells. These are mostly utilitarian shells and not really damage dealers. The various shells can be read about in the design document. 
 
- Seclathe still has incendiary slugs for lings. Autolathe still has beanbags for the bartender and security. Cargo can still order rubbershot and beanbag shot in the ammo crate.  You can actually use a wrench on your riot blasters to let them use normal shotgun shells (at a misfire risk). This is mostly to help tide people over for a bit before taking a look at stuff like armor and immunities. At that point, I think we can actually just toss riot shotguns completely.
 
- Cargo can also order voidshot mostly ported over from here https://github.com/tgstation/tgstation/pull/67250. Shaps said it was okay to steal. It's not completely the same, significantly less damage being the biggest difference, the other being that it's an energy shell.

### Deltra

Cargo can also order Deltra spool-up automatic energy guns with built in forcefield projectors. They're real shooty. Read about them in the document.

Sprites are a work in progress but here is a picture of the Delta it's a big:
deltra inhand

Of all the things that are likely to be 'too much', this is probably it. But I'm curious to see how it works as a crew weapon ~~before tossing it to ERTs~~.

## Why It's Good For The Game

Read my "Why it's Good for the Game" for these changes here: https://github.com/tgstation/tgstation/pull/64280

Please read the design document: https://hackmd.io/@6DbtsAKCTtW_9MByKFjZqg/r1xYKCNOt

## What if it doesn't work and the game is RUINED?

Don't worry about it.

## Changelog

:cl: Necromanceranne for code and sprites, quin for sprites
add: Riot Blaster Shotgun (replacing all but one riot shotgun and combat shotguns). Found in cargo and the armory.
add: Deltra automatic energy cannon. Found in cargo.
add: Energy shotgun shells. They blink and also do energy things. There are a good few of them.
del: DRAGnets were replaced with energy shell equivalents you can print from the security protolathe.
balance: The technology shells have been changed quite a bit. Many of them are now energy shotgun shells. Some no longer exist for crew (or at all). Who cares about dragonsbreath. Not me. So it still exists. Whatever.
balance: You can print beanbags in the autolathe. Security protolathe can still print incendiary slugs, but not other shotgun shells. For changelings, obviously. You can get rubbershot and beanbags from cargo packs.
balance: You should really just read the document outlined here for more information: https://hackmd.io/@6DbtsAKCTtW_9MByKFjZqg/r1xYKCNOt
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
Hopefully the Deltra can be our answer to sec being reliant on the stun baton for crowd control.

**Comment by Palpetinus** [comment]
I get the idea behind the design document, but I can already predict that at some point down the line, nukies and tots will win like 90% of encounters simply by EMPing before every fight. If the average seccie then has no tot gear themselves (somehow) or is powergaming a butcher's cleaver and such, they've already lost. I will watch this with great, antagonistic interest.

**Comment by MTandi** [comment]
I don't really care about the idea behind this, but I'm completely against removal of ballistic guns. 
They're loud, fun, and a symbol of rough power. 
Balance them out instead of replacing if you don't like something.

**Comment by Wallemations** [comment]
I don't get the point of making it an energy weapon if you still need to reload it with shells

File: code/datums/components/fullauto.dm [hunk 1]
```diff
@@ -233,6 +243,10 @@
 		return FALSE
 	shooter.face_atom(target)
 	var/next_delay = autofire_shot_delay
+	if(windup_autofire)
+		next_delay = clamp(next_delay - current_windup_reduction, round(autofire_shot_delay * windup_autofire_cap), autofire_shot_delay)
+		current_windup_reduction = (current_windup_reduction + round(autofire_shot_delay * windup_autofire_reduction_multiplier))
+		timerid = addtimer(CALLBACK(src, .proc/windup_reset, FALSE), 3 SECONDS, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_STOPPABLE)
```
Comment (Fikou) [inline code/datums/components/fullauto.dm:249]: can you make the 3 seconds a variable?

File: code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 2]
```diff
@@ -118,6 +118,94 @@
...
+	can_misfire = FALSE
+	misfire_probability = 0
+	misfire_percentage_increment = 25
+	projectile_damage_multiplier = 0.50
+
+	///Our cell for keeping our shotgun charged
+	var/obj/item/stock_parts/cell/cell
+	var/cell_type = /obj/item/stock_parts/cell
+
+	///Vars for tracking power expenditure
```
Comment (Fikou) [inline code/modules/projectiles/guns/ballistic/shotgun.dm:144]: this isnt how you should use autodocs, look at other files

File: code/modules/projectiles/guns/energy/laser.dm [hunk 3]
```diff
@@ -191,3 +175,46 @@
...
+	forcebarrier = new /obj/item/forcefield_projector/fast(src)
+
+/obj/item/gun/energy/laser/deltra/examine(mob/user)
+	. = ..()
+	if(forcebarrier) //just in case something happens
+		. += span_notice("Right-click to use the forcefield projector.")
+		. += span_notice("Alt-click to drop the forcefield barriers.")
+		. += span_notice("It is currently sustaining [LAZYLEN(forcebarrier.current_fields)]/[forcebarrier.max_fields] fields, and it's [round((forcebarrier.shield_integrity/forcebarrier.max_shield_integrity)*100)]% charged.")
+
+/obj/item/gun/energy/laser/deltra/AltClick(mob/user)
```
Comment (Fikou) [inline code/modules/projectiles/guns/energy/laser.dm:209]: why not have this be when you right click the weapon?

**Comment by Shadowflame909** [comment]
I thought I saw something mentioning the warden but I dont see it. Does this do away with the wardens gun too?

**Comment by VladinXXV** [comment]
>In addition, the stun baton is still as reliable as ever when combatting foes only immune to ranged threats (like sleeping carp and deswords against beams).

So has anyone told you about the desword having a 75% melee block chance? Or that it can drop you in like, 2 hits (1 is crippling considering wounds)? Would you like to volunteer yourself as the sacrifice who tries to baton the desword-noslip user? Maybe not quite "immune" but it is certainly a suicide mission. Also kinda applies to a sleeping carp user, since they'll just steal your baton.

**Comment by thgvr** [comment]
> I thought I saw something mentioning the warden but I dont see it. Does this do away with the wardens gun too?

Warden has not had their special shotgun in quite some time. I don't know why you think they still have it.

Anyways, excited to try this out.

**Comment by anclandor** [comment]
> Pretty sure the reason why NanoTrasen doesn't want to issue ballistics to their stations is because they're pretty destructive. They could probably penetrate through the hull and causes breaches, which would be pretty hard to maintain. Energy weapons are much safer and healthier for the station.

A. This is frankly untrue.  Ballistics don't blow through walls any faster than laser guns, and every window on the station is reinforced, so they don't cause much window damage.  Nuke ops do not even destroy them on accident often.

B. Since when has NT cared more about employee safety than cost cutting.  It is totally out of character for them to buy more expensive laser weapons to protect the crew.

**Comment by castawaynont** [comment]
cool minigun

File: code/datums/components/fullauto.dm (hunk 1 above)
Comment (necromanceranne) [reply code/datums/components/fullauto.dm:249]: Ye

File: code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 2 above)
Comment (necromanceranne) [reply code/modules/projectiles/guns/ballistic/shotgun.dm:144]: Damnit I've been deceived from like a whole year ago

File: code/modules/projectiles/guns/energy/laser.dm (hunk 3 above)
Comment (necromanceranne) [reply code/modules/projectiles/guns/energy/laser.dm:209]: I literally didn't think that was a thing. Nice.

**Comment by ATH1909** [comment]
image
This isn't something to be brushed over- dragNETs are good because, mechanics-wise, they're closer to disablers than shotguns- they're the same size as a disabler (normal-sized) and they use the 0.2 second (IIRC) energy gun/disabler shot cooldown instead of the horrendous 0.8 second(!) shotgun one. Forcing them to use shotgun mechanics will absolutely obliterate their viability- you'll be effectively **quartering** their DPS and making carrying one alongside a secbelt deny you access to ranged lethals (unless you fumble around for 10 billion years to try to swap out your shotgun shells).

I'm also very, very concerned that this transition over to pure energy weapons will make sec unhealthily vulnerable to emps. Without ballistics, emp'd sec officers have... bolas and flickering batons? Maybe combat kni- nope, those got nerfed into the ground. Lit welders, I guess? Regardless, I think this means that emps will kind of unavoidably insta-fuck all officers in a pretty huge AoE, especially if they're fighting stun-immune creatures that a flickering baton won't work on at all (hulks, druggies, heretic stalkers, emp strain blobs, etc.).

**Comment by GuillaumePrata** [comment]
> Would you like to volunteer yourself as the sacrifice who tries to baton the desword-noslip user? Maybe not quite "immune" but it is certainly a suicide mission. Also kinda applies to a sleeping carp user, since they'll just steal your baton.

Non tackling enjoyer confirmed

**Comment by YuliceTheDruid** [comment]
> Non tackling enjoyer confirmed

Tackling can fuck over a DEsword user but it's still RNG, even if it's heavily stacked in your favor. And if you miss the tackle you're also just instantly dead.

**Comment by GuillaumePrata** [comment]
> Tackling can fuck over a DEsword user but it's still RNG, even if it's heavily stacked in your favor. And if you miss the tackle you're also just instantly dead.

Still better RNG than batoning a DEsword/Carp user.
My point was that sec has roundstart gear to deal with these threats already without needing ballistic to counter them, people are just too scared to use it.

**Comment by Ryll-Ryll** [comment]
I still don't agree that replacing ballistics with energy weapons is an improvement, in large part because I think bullet wounds are more engaging (pierce wounds >>> burn wounds in terms of design quality, bleeding is a much more developed mechanic than burn infections), but also just in terms of variety and presentation. I also don't think misfire chances on guns are a great form of RNG that should be expanded to more things, since in my experience it just makes people avoid using anything that can fail entirely (like using .357 in the .38).

The Deltra is awfully interesting though, especially with the ability to drop barriers on demand for defensive purposes. I think that'd be a very good addition

**Comment by Fikou** [comment]
> I still don't agree that replacing ballistics with energy weapons is an improvement, in large part because I think bullet wounds are more engaging (pierce wounds >>> burn wounds in terms of design quality, bleeding is a much more developed mechanic than burn infections)

gravity and voidshots are brute, energy =/= burn necessarily

**Comment by wesoda25** [comment]
What is the point behind the 5 second charge for the shotgun? It seems unintuitive and annoying. 

I also don't understand why you would want to keep energy armor rare if your goal is to make it the most commonly available source of ranged damage on the station.

**Comment by BlueMemesauce** [comment]
> Ballistics just feel more fun to shoot

honestly always felt the opposite, ballistics feel clunky and slow to shoot while energy guns are smooth and fast

**Comment by necromanceranne** [comment]
> I still don't agree that replacing ballistics with energy weapons is an improvement, in large part because I think bullet wounds are more engaging (pierce wounds >>> burn wounds in terms of design quality, bleeding is a much more developed mechanic than burn infections), but also just in terms of variety and presentation. I also don't think misfire chances on guns are a great form of RNG that should be expanded to more things, since in my experience it just makes people avoid using anything that can fail entirely (like using .357 in the .38).
> 
While I think I've proven that wounds especially can exist with modularity even with energy weaponry thanks to cryo pistols and even the Deltra + voidshot, I am rethinking the misfire for the alt ammo. Frankly, there is no need for it. I'll just remove it and let the riot blaster shoot the alt ammo with no restrictions once altered.

**Comment by ATH1909** [comment]
> That way, if you did want to use any normal shotgun shells with it, or have access to it, the riot blaster can utilize it.

Didn't you remove the ability to print most normal shotgun shells, though?

**Comment by SkeletalElite** [comment]
Gonna need more explanation on why the dragnets are being removed. 

Honestly don't care about the fact that as the current state of the document has the weapons pretty much completly useless compared to just using the disabler because the one 1 riot shotgun still exists for the scenarios that require it because I can choose to just not use them, but I don't see why the dragnet is being removed its one of my favorite tools as sec.

**Comment by necromanceranne** [comment]
> Gonna need more explanation on why the dragnets are being removed.
> 
> Honestly don't care about the fact that as the current state of the document has the weapons pretty much completly useless compared to just using the disabler because the one 1 riot shotgun still exists for the scenarios that require it because I can choose to just not use them, but I don't see why the dragnet is being removed its one of my favorite tools as sec.

They're outdated as fuck. I think they used to be from when secborgs were first merged.

They function as one handed energy gun shotguns. They can be dualwielded and bagged. They have been ignored almost the entire time they've existed.

They aren't gone functionally. The shells that utilize the dragnet's projectiles still exist and are printable. The TRIPnets even have some free boxes in the armory ready for use.

**Comment by SkeletalElite** [comment]
> They're outdated as fuck. I think they used to be from when secborgs were first merged.

Then update them?

> They function as one handed energy gun shotguns. They can be dualwielded and bagged. They have been ignored almost the entire time they've existed.

because their primary function is literally only to slow people down. I don't see a problem with them being bagged, I've literally never seen someone dual wield them, I don't imagine it being excessively strong though.

> They aren't gone functionally. The shells that utilize the dragnet's projectiles still exist and are printable. The TRIPnets even have some free boxes in the armory ready for use.

No one is going to use the new tripnets because they have to haul around a bulky object when they could just get a box of energy bolas instead.

**Comment by Miczu555PL** [comment]
> because their primary function is literally only to slow people down.

Any slowdown is crippling in a fight.

> No one is going to use the new tripnets because they have to haul around a bulky object when they could just get a box of energy bolas instead.

I'd rather dodge 7 bolas than not be able to get close at all.

**Comment by Palpetinus** [comment]
> I still don't agree that replacing ballistics with energy weapons is an improvement, in large part because I think bullet wounds are more engaging

I was just about to mention the same, but thermal pistols use purely just energy to fire, while their bullets cause pierce wounds. So Energy doesn't automatically mean burn.

I think what sec needs is Mass Effect-ish style gauss weaponry, that in flavor electromagnetically accelerates tiny bits of superdense materials (lead, uranium). So mechanically what the thermal pistols do, but in "good for firefights with nukies"-rifle form.

**Comment by SinfulBliss** [comment]
This is cool but the "why it's good for the game" section being redirected to thermal pistols doesn't apply to shotguns as it did to autorifles.

The main reason autorifles got knocked was because you had essentially infinite ammo with fast reloads and no need to retreat. Shotguns only have 7 shots and ammo is very annoying to lug around so you'll probably need to retreat after. It's also incredibly slow to reload.

**Comment by necromanceranne** [comment]
I changed voidshot to do 15 damage total, but now it can randomly do all three kinds of brute based wounds. Blunt, slash and puncture. This should ultimately result in more mangled bodies rather than outright dead bodies. They don't lose damage in-flight and have a naturally pretty tight cone, but they do lose wounding power.

Riot blaster doesn't misfire anymore with alternative ammo, that was a bad idea honestly.

**Comment by anclandor** [comment]
Could you add one Deltra to the armory?  I feel like this would encourage sec to actually learn how to use it rather than it just being something that is there but never bought, because nobody has practiced with it and when the revs come knocking at your door they already own cargo.  It also doesn't seem like it is massively powerful, so I don't think giving sec one round start would be unbalanced.

**Comment by necromanceranne** [comment]
Nah.

**Comment by realthomas69** [comment]
dont remove dragnets

**Comment by ANAME593** [comment]
> I changed voidshot to do 15 damage total, but now it can randomly do all three kinds of brute based wounds. Blunt, slash and puncture. This should ultimately result in more mangled bodies rather than outright dead bodies. They don't lose damage in-flight and have a naturally pretty tight cone, but they do lose wounding power. Their wounding power point blank is _very high_.

if voidshot is only accessible via cargo then it should be somewhat viable, printing shells from the roundstart lathe (which is gonna be loaded with mats)  is much easier to do then say going to cargo and buying your ammo(which means spending cash)  waiting 60 seconds then loading your gun with your new found ammo. if you HAVE TO buy voidshot shells from cargo in order to use them then make them do atleast 20 damage because having "high wound power" doesn't mean much when there are so many ways to treat a wound before it cucks you. and people aren't gonna use them if there is a much better option waiting for them in the sec lathe which as i said is available roundstart & will have a ton of mats

**Comment by Palpetinus** [comment]
Cargo this, lathe that..
How about giving sec some actual research for once and locking it behind research into the lathe? The only things that ever get printed are speed loaders and forensic scanners, one being available roundstart and the other not actually being sec research.
Doctors get T2 tools (well.. not really, anymore), Botany gets its bio-whatchamacallit shears, why can't sec get T2 weaponry from its lathe?

**Comment by necromanceranne** [comment]
Well, they do. It's the gun kits. But I mean, a gunkit for deltra isn't an outrageous suggestion. But I'm not super sure people will like that. I think the gun kits haven't been as successful as I had hoped. Might be worth revisiting, even by just putting this pr aside and looking at how to fix that.

**Comment by necromanceranne** [comment]
Gonna put this one back into the oven, break it apart a bit and pr something later. This might not be the best idea right now, but the individual functionality stuff can be useful for other stuff.

@Ryll-Ryll would you like the Deltra as something separate from this pr? If so, I demand blood tithe. Name your sacrifice. (it can be small) (it can be nothing)

**Comment by Shadowflame909** [comment]
This is so sad

Another good feature ended by peopple not wanting the game to change

**Comment by necromanceranne** [comment]
It's not that (necessarily). The idea just needs some more work overall to get to a good state and maybe with some considerations elsewhere.

**Pull request closed without merging**